### ISSUE TITLE
Add Val Loss Calculation Loop

### ISSUE THREAD

**Issue by harshagrawal13** [atom-51/jespr#3]
Add validation loss metrics such as loss, etc.
